test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,633 @@
+            self.assertEqual(first_message['customer_guid'], self.valid_customer_guid,
+                             "Customer GUID in response does not match the input customer GUID")
+
+        # Check for reverse chronological order
+        for i in range(len(messages) - 1):
+            current_timestamp = messages[i]['timestamp']
+            next_timestamp = messages[i + 1]['timestamp']
+            self.assertGreaterEqual(current_timestamp, next_timestamp,
+                                    "Messages are not in reverse chronological order")
+
+        logger.info("=== Test Case 4 Completed ===\n")
+
+    def test_retrieval_with_page_two(self):
+        """Test case 5: Valid customer_guid, chat_id, page=2, and page_size=10"""
+        logger.info("=== Starting Test Case 5: Valid customer_guid, chat_id, page=2, and page_size=10 ===")
+
+        # Assuming we have enough messages to test pagination
+        for page in range(1, 4):  # Test first three pages
+            url = f"{self.BASE_URL}/getallchats"
+            data = {
+                "customer_guid": self.valid_customer_guid,
+                "chat_id": self.valid_chat_id,
+                "page": page,
+                "page_size": 5
+            }
+            logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+            response = requests.post(url, json=data)
+            logger.info(f"OUTPUT: Response status code: {response.status_code}")
+            logger.info(f"OUTPUT: Response content: {response.text}")
+
+            if response.status_code == 200:
+                messages = response.json().get('messages')
+                self.assertIsInstance(messages, list)
+                self.assertLessEqual(len(messages), 5, "Messages returned exceed page size")
+            else:
+                self.assertEqual(response.status_code, 404, "Expected 404 for no more messages")
+
+            logger.info("=== Test Case 5 Completed ===\n")
+
+    def test_single_message_retrieval(self):
+        """Test case 6: Valid customer_guid, chat_id, page=1, and page_size=1"""
+        logger.info("=== Starting Test Case 6: Valid customer_guid, chat_id, page=1, and page_size=1 ===")
+
+        # Step 1: Create an initial chat and get the chat_id
+        initial_chat_data = {
+            "customer_guid": self.valid_customer_guid,
+            "question": "You will get the correct answer once AI is integrated."
+        }
+        chat_url = f"{self.BASE_URL}/chat"
+        chat_response = requests.post(chat_url, json=initial_chat_data)
+        self.assertEqual(chat_response.status_code, 200, "Failed to create initial chat.")
+        self.valid_chat_id = chat_response.json().get("chat_id")
+        logger.info(f"Created initial chat with chat_id: {self.valid_chat_id}")
+
+        # Step 2: Add a chat message to the chat
+        message_to_add = "You will get the correct answer once AI is integrated."
```

Comment:
This should again be a question

Reply:
I changed the question in this testcase, sir.